Fix skl_dram_get_dimm_info warning message in Tiger Lake controller

Alexleong223 opened the pull request:
Up to now we were reading some DRAM information from MCHBAR register
and from pcode what is already not good but some GEN12(TGL-H and ADL-S)
platforms have MCHBAR DRAM information in different offsets.

This was notified to HW team that decided that the best alternative is
always apply the 16gb_dimm watermark adjustment for GEN12+ platforms
and read the remaining DRAM information needed to other display programming from pcode.

So here moving the DRAM pcode function to intel_dram.c, removing
the duplicated fields from intel_qgv_info, setting and using
information from dram_info.

v2:
- bring back num_points to intel_qgv_info as num_qgv_point can be
overwritten in icl_get_qgv_points()
- add gen12_get_dram_info() and simplify gen11_get_dram_info()

Reviewed-by: Lucas De Marchi <lucas.demarchi@intel.com>
Signed-off-by: José Roberto de Souza <jose.souza@intel.com>
Link: https://patchwork.freedesktop.org/patch/msgid/20210128164312.91160-2-jose.souza@intel.com
(cherry picked from commit 5d0c938)

[Leong Kai Yean: dropped call to intel_pcode_init() not present in 5.10]
Signed-off-by: Leong Kai Yean <alex.leong@ni.com>

zehortigoza committed:
drm/i915/gen11+: Only load DRAM information from pcode

Up to now we were reading some DRAM information from MCHBAR register
and from pcode what is already not good but some GEN12(TGL-H and ADL-S)
platforms have MCHBAR DRAM information in different offsets.

This was notified to HW team that decided that the best alternative is
always apply the 16gb_dimm watermark adjustment for GEN12+ platforms
and read the remaning DRAM information needed to other displayprogramming from pcode.

So here moving the DRAM pcode function to intel_dram.c, removing
the duplicated fields from intel_qgv_info, setting and using
information from dram_info.

v2:
- bring back num_points to intel_qgv_info as num_qgv_point can be
overwritten in icl_get_qgv_points()
- add gen12_get_dram_info() and simplify gen11_get_dram_info()

Reviewed-by: Lucas De Marchi <lucas.demarchi@intel.com>
Signed-off-by: José Roberto de Souza <jose.souza@intel.com>
Link: https://patchwork.freedesktop.org/patch/msgid/20210128164312.91160-2-jose.souza@intel.com
(cherry picked from commit 5d0c938ec9cc96fc7b8abcff0ca8b2a084e9c90c)

[Leong Kai Yean: dropped call to intel_pcode_init() not present in 5.10]
Signed-off-by: Leong Kai Yean <alex.leong@ni.com>

bstreiff commented:
If this is a cherry-pick of an upstream commit, you should preserve the SBOs and append yours to the end, e.g.:

```
Reviewed-by: Lucas De Marchi <lucas.demarchi@intel.com>
Signed-off-by: José Roberto de Souza <jose.souza@intel.com>
Link: https://patchwork.freedesktop.org/patch/msgid/20210128164312.91160-2-jose.souza@intel.com
[Leong Kai Yean: dropped call to intel_pcode_init() not present in 5.10]
Signed-off-by: Leong Kai Yean <alex.leong@ni.com>
```

Alexleong223 commented:
> If this is a cherry-pick of an upstream commit, you should preserve the SBOs and append yours to the end, e.g.:
> 
> ```
> Reviewed-by: Lucas De Marchi <lucas.demarchi@intel.com>
> Signed-off-by: José Roberto de Souza <jose.souza@intel.com>
> Link: https://patchwork.freedesktop.org/patch/msgid/20210128164312.91160-2-jose.souza@intel.com
> [Leong Kai Yean: dropped call to intel_pcode_init() not present in 5.10]
> Signed-off-by: Leong Kai Yean <alex.leong@ni.com>
> ```

@bstreiff 
Updated according to your comment.

gratian commented:
Fixed line breaks in the commit message and pulled as commit 1cbdb3c696d86c12c08b622262b49531a78925e3 ("drm/i915/gen11+: Only load DRAM information from pcode") in nilrt/master/5.10.
